### ISSUE TITLE
Add compose support for static IP addressing

### ISSUE THREAD

**Issue by Dyson228** [IceWhaleTech/CasaOS#1324]
When using a MacVLAN network, there is an additional option that would support assigning a static IP address.  In the yaml file it is:

Services:
  networks:
    <macVLAN Network>:
       ipv4_address: <static IP>

Currently, if I edit the existing docker-compose file to add this line CasaOS displays my app as "legacy", and I lose the ability to edit the compose settings through the web UI.

Additionally, this isn't working to give me a static IP address.  I'm still troubleshooting that.  I believe it may be related to compose version, but I haven't found what version compose CasaOS is using by default. 

Right now, my alternative is to define a MacVLAN network with a /32 address, which I may do as this is likely the only contain I will have running on that VLAN.  I do think having an entire VLAN defined as a network with a static IP is a cleaner solution though.

**Comment by ETWang1991** [comment]
Got it

**Comment by kingfish65** [comment]
Is there any update on macvlan support?

**Comment by dimisc** [comment]
Yes! Would love this functionality. I am setting a small server I can take with me when travelling. I want to set internal static IPs to my docker containers so that they are network agnostic and continue to function and communicate with each other when the server is connected to a new network with DHCP.

**Comment by Dankni95** [comment]
Why isnt this yet possible?

**Comment by duhmojo** [comment]
I'm just enjoying Casa now, but missing macvlan, network customization, is a problem for me. I might come back at some point but seeing requests for it going back to 2022 tells me the project has other priorities.

**Comment by renedis** [comment]
+1. I now continuesly need to set it manually in the "var/lib/casaos/apps" folders and set it up after each restart in the "docker-compose.yml" files.

```
        networks:
            vlanx:
                ipv4_address: x.x.x.x
networks:
    default:
        name: xxxx_xxxx
    vlanx:
        name: vlanx
        external: true
```
